### PR TITLE
[sc-2972] feat: Show modal on correct position

### PR DESCRIPTION
**Pull request by gvgesanto2**
## Describe the changes in this pull request

This PR makes the following changes:
- Shows the ``New Glossary dialog`` on the correct position.

### CODE REVIEW

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js [hunk 1]
```diff
@@ -1,67 +1,67 @@
...
+				const glossaryTermExpression = glossaryModal.querySelector( '#glossary_term_expression' );
+				const glossaryTermDefinition = glossaryModal.querySelector( '#glossary_term_definition' );
 
-				const range = editor.model.document.selection.getFirstRange()
-				let selection = "";
-				for (const item of range.getItems()) {
-					if (item.data) {
+				const range = editor.model.document.selection.getFirstRange();
+				let selection = '';
+				for ( const item of range.getItems() ) {
```
Comment (diego-aslz) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js]: Can you fix your formatting? Why the extra spaces in several lines? Is this your editor?

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js [hunk 2]
```diff
@@ -1,67 +1,67 @@
...
 						selection += item.data;
 					}
 				}
 
-				if (selection.length) {
+				if ( selection.length ) {
 					// WE USE `window.getSelection()` BECAUSE EDITOR SELECTION DOESN'T ACCEPT `getBoundingClientRect`
-					const position = window.getSelection().getRangeAt(0).getBoundingClientRect();
-					// SET BODY TO FOCUS MODE AND SHOW MODAL ON CORRECT POSITION
-					document.body.classList.add("focus-mode");
```
Comment (NicholasJarr) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js:38]: Is `focus-mode` not necessary anymore?

**Review by NicholasJarr — COMMENTED** [review]
Can you detail a little more why the previous position was wrong?

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js (hunk 1 above)
Comment (gvgesanto2) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js]: Hi @diego-aslz! It's the [.eslintrc.js](https://github.com/Helpjuice/ckeditor5/blob/helpjuice-build-v2/.eslintrc.js) and [.editorconfig](https://github.com/Helpjuice/ckeditor5/blob/helpjuice-build-v2/.editorconfig) files that are causing those formatting issues. Maybe we should update them to the same config we use in the helpjuice repo.

Comment (diego-aslz) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js]: This is eslint then, our editorconfig is not setup like this. Can we fix the linter?

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js (hunk 2 above)
Comment (gvgesanto2) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js:38]: Hi @NicholasJarr! That's correct, I removed the ``focus-mode`` class from the app. That's why I'm removing it here too.

About the previous position, it seemed to get misplaced when the user scrolled down the page as we can see in the video below:

https://github.com/Helpjuice/ckeditor5/assets/64566209/ebfd6bce-51bd-4054-975c-abb61dd9735c

The new position update shows the glossary dialog always right below the selected text and inside the editor body:

image

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js (hunk 1 above)
Comment (gvgesanto2) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js]: Yeah, sure! The extra spaces were caused by Eslint's ``space-in-parens`` rule. I turned it off and fixed the formatting! 

About the .editorconfig, it's set up differently from the configuration used in the ``helpjuice4`` repo. We are using a formatting of 2 spaces there while in this repo we're using tabs with a tab width of 4.

Maybe we should keep the same configuration on both. What do you think @diego-aslz?

Comment (diego-aslz) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js]: I think that's a good idea. Up to you if you wanna do that separately, though.

**Review by diego-aslz — APPROVED** [review]

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js (hunk 2 above)
Comment (NicholasJarr) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/glossary/glossaryui.js:38]: Thanks for the explanation! 👍🏻

**Review by NicholasJarr — APPROVED** [review]

**Comment by NicholasJarr** [comment]
I will close this PR since it's stale. If we decide to get back to it, we can always reopen or use it as reference.

**Pull request closed without merging**